File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let sLen = 0;
  let eLen = numbers.length - 1;

  while (sLen < eLen) {
    let currentSum = numbers[sLen] + numbers[eLen];
    if (currentSum == target) {
      return [sLen+1, eLen+1];
    } else if (currentSum < target) {
      sLen++;
    } else eLen--;
  }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int slen = 0;
        int rlen = numbers.Length - 1;

        while (slen<rlen)
        {
            int currentsum = numbers[slen] +  numbers[rlen] ;
            if (currentsum == target)
                return new int[]{slen+1,rlen+1};
            else if (currentsum<target)
                slen++;
            else
                rlen--;
        }
        return new int[]{slen+1,rlen+1};
    }
}